src/generate-query.ts

```diff
@@ -509,9 +506,52 @@ function getMissingSlicingArg(
   if (usesSlicingArg) return null
 
   // Return the first slicing arguments:
-  return field.arguments.find((arg) =>
-    slicingArguments.includes(arg.name.value)
+  return field.arguments.find((arg) => {
+    return slicingArguments.find((slicingArgument) => {
+      let ret = getMissingSlicingArgHelper(
+        slicingArgument.split('.'),
+        arg,
+        schema
+      )
+      if (ret) return arg
+    })
+  })
+}
+
+function getMissingSlicingArgHelper1(
+  tokenizedPath: string[],
+  type: GraphQLNamedType,
+  schema: GraphQLSchema
+) {
+  if (type?.astNode?.kind === Kind.INPUT_OBJECT_TYPE_DEFINITION) {
+    return type.astNode.fields.find((field) => {
+      let ret = getMissingSlicingArgHelper(
+        tokenizedPath.slice(1),
+        field,
+        schema
+      )
+      if (ret) return ret
```

Comment:
You can just return the result. I don't think you need to check if it exists.

Reply:
sounds good